Added post-survey materials form for CSF, and cleaned up the pd-workshop-survey.

aoby opened the pull request:
1. Addressed remaining feedback from https://github.com/code-dot-org/code-dot-org/pull/8475
   1. Moved haml `:javascript` into pd-workshop-survey.js
   2. Converted multi-select to checkboxes:
      ![image](https://cloud.githubusercontent.com/assets/8405316/15596446/6b7d71be-237c-11e6-8bc4-6879ac2f55b7.png)
2. Added post-survey materials shipping address form for CSF, as outlined in the [spec](https://docs.google.com/document/d/1HPbcOIEKqrd2p7s-qHMveg3ovOWmg6lftcKck-Bn2Rs).

tanyaparker commented on pegasus/forms/pd_workshop_materials.rb:
```diff
@@ -0,0 +1,24 @@
+    result[:state_s] = required stripped data[:state_s]
+    result[:zip_code_s] = required stripped data[:zip_code_s]
+    result[:email_s] = required stripped data[:email_s]
+    result[:phone_number_s] = required stripped data[:phone_number_s]
```
Do we care about the format? Does stripped also get rid of () or - ? Do we only want integers?

aoby replied:
`stripped` just strips whitespace off the ends. It will remain a string and retain any original formatting. What do you think? Should we require only numbers, or add a phone number helper and allow xxx-xxx-xxxx or (xxx)xxx-xxxx, or just leave as open string?

ashercodeorg replied:
Assuming we are wanting to accept US phone numbers only, I vote for forcing a specific format from the user. It'll help prevent typos, later parsing on our side, etc.

tanyaparker replied:
True, this probably won't be used internationally, at least for several more years. Still would prefer all integers that you can format in the views/emails later, but at the very least force them to all have the same US format.

tanyaparker commented on pegasus/forms/pd_workshop_materials.rb:
```diff
@@ -0,0 +1,24 @@
+    result[:city_s] = required stripped data[:city_s]
+    result[:state_s] = required stripped data[:state_s]
+    result[:zip_code_s] = required stripped data[:zip_code_s]
+    result[:email_s] = required stripped data[:email_s]
```
There are `zip_code` and `email_address` form helpers, should we use those here?

aoby replied:
Thanks. Yes, I'll use those.

tanyaparker commented:
Just 1 question, otherwise LGTM

aoby commented:
Updated with the following:
1. Validation for phone number, zip code, and email. 
   
   I added a couple methods in /lib extending a US phone number regex we already had there. Now phone number will validate the format and save as just a string of 10 digits.
2. Removed plp conditional on 2 questions. Now they show up for everyone (spec [change](https://docs.google.com/document/d/1HPbcOIEKqrd2p7s-qHMveg3ovOWmg6lftcKck-Bn2Rs))
3. Converted single-item agree scale into a normal single-select.
4. Switched single-select view from a dropdown to radio buttons. This last decision is still being discussed over email, so we may back it out.

tanyaparker commented on pegasus/sites.v3/code.org/public/pd-workshop-survey/materials/splat.haml:
```diff
@@ -0,0 +1,105 @@
+
+    .form-group
+      %label.control-label{for: 'phone_number_s'}
+        Phone Number
```
Not sure what it looks like in the view but could there be a little suggestion/placeholder text like XXX-XXX-XXXX or 123-456-7890? Even though we'll be smart to accept pretty much any format they try, might be easier to be directed to think one way?

aoby replied:
Added a placeholder:

![image](https://cloud.githubusercontent.com/assets/8405316/15756062/6761712e-28b4-11e6-8085-dae2cc028e9d.png)

tanyaparker commented:
Nice solution! LGTM

aoby commented:
I realized that I broke functionality in the old `RegexpUtils.find_potential_phone_number` code, which was caught by another test. I fixed that and added a unit test for that specific function. Please take another look.

tanyaparker commented:
Looks good!

bcjordan commented on shared/test/test_regexp.rb:
```diff
@@ -0,0 +1,53 @@
+require 'cdo/regexp'
+class RegexpUtilsTest < Minitest::Test
+  EXPECTED_DIGITS = '1234567890'
+  VALID_US_PHONE_NUMBERS = %w[
```
Oh, funny! Looks like there is another test file for this in https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/test/test_regexp_utils.rb#L7-L7 (think that was before shared existed, though)

aoby replied:
Thanks Brian. Funny I didn't find that one. I chatted with @tanyaparker and we prefer the new regex that will accept phone numbers as just a string of digits. I'll update this test.

bcjordan replied:
@tanyaparker @aoby hold on, this is currently used for avoiding letting users share programs with personally identifiable information. That would mean programs with strings of digits would count as containing phone numbers, and would then show an error on share (e.g. a student saying "I would like to withdraw 1000000000 dollars please" in their program). It sounds like we might want two different utilities in this case.

aoby replied:
Thanks @bcjordan. Here you go: https://github.com/code-dot-org/code-dot-org/pull/8746
